xbmc/threads/platform/RecursiveMutex.h

```diff
@@ -0,0 +1,63 @@
+/*
+ *      Copyright (C) 2005-2013 Team XBMC
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#pragma once
+#include <mutex>
+
+#if (defined TARGET_POSIX)
+#include <pthread.h>
+namespace XbmcThreads
+{
+  // forward declare in preparation for the friend declaration
+  class RecursiveMutex
```

Comment:
coding guidelines -> CRecursiveMutex

Reply:
https://github.com/xbmc/xbmc/pull/13454/files#diff-363ac52c76554888a80f57f4569573d4L37

Reply:
this is new code now and I don't see any reason why it should not comply to our coding standards